evm/src/arithmetic/modular.rs

```diff
@@ -246,21 +270,41 @@ fn generate_modular_op<F: RichField>(
     let aux_limbs = pol_remove_root_2exp::<LIMB_BITS, _, { 2 * N_LIMBS }>(constr_poly);
 
     lv[MODULAR_OUTPUT].copy_from_slice(&output_limbs.map(|c| F::from_canonical_i64(c)));
-    lv[MODULAR_OUT_AUX_RED].copy_from_slice(&out_aux_red.map(|c| F::from_canonical_i64(c)));
-    lv[MODULAR_QUO_INPUT].copy_from_slice(&quot_limbs.map(|c| F::from_noncanonical_i64(c)));
-    lv[MODULAR_AUX_INPUT].copy_from_slice(&aux_limbs.map(|c| F::from_noncanonical_i64(c)));
-    lv[MODULAR_MOD_IS_ZERO] = mod_is_zero;
+
+    for (i, lo, j, hi) in izip!(
+        MODULAR_QUO_INPUT_LO,
+        quot_limbs[..N_LIMBS]
+            .iter()
+            .map(|&c| F::from_noncanonical_i64(c)),
+        MODULAR_QUO_INPUT_HI,
+        quot_limbs[N_LIMBS..]
+            .iter()
+            .map(|&c| F::from_noncanonical_i64(c)),
+    ) {
+        lv[i] = lo;
+        nv[j] = hi;
+    }
```

Comment:
Nit: I think this would be more readable as two separate loops, but granted, it is going to be faster as one loop.